content/push/publish.textile

```diff
@@ -0,0 +1,768 @@
+
+Publishing via channels is modeled on Ably’s "channel":#channels infrastructure, facilitating the delivery of push notifications across a network of subscribed devices. This method publishes messages through predefined channels, which devices must "subscribe":#sub-channels to to receive updates. This process ensures registered devices in the specified channels recieve the correct push notifications. Publishing via channels is particularly useful for publishing notifications to multiple groups with varying privileges.
+
+h3(#sub-channels). Subscibe to channels
+
+Subscribe to Ably "channels":/channels only if you are publishing push notifications via channels method. You can subscribe using either "@deviceId@":/push/publish#sub-deviceID
+ or "@clientId@":/push/publish#sub-clientID , offering flexibility. These options enable granular control _by subscribing to devices with specific *@deviceIds@*_ and support for broader operations _by subscribing to all devices associated with a @clientId@_.
+
+h3(#sub-deviceID). Subscibe to channels using @deviceId@
+
+A @deviceId@ is a unique identifier assigned to a device when it first registers with Ably. 
+
+If your application (the client) has the capability to subscribe to push notifications using the @push-subscribe@ "capability":/auth/capabilities, you can directly subscribe to these notifications for a device. This approach allows you to target specific devices with push notifications, ensuring that only the intended recipients receive the messages.
+
+There are two ways to subscribe by @deviceId@: *With* and *without* a @push-subscribe@ capability.
+
+h4. With push-subscribe capabilities
+
+If your application has @push-subscribe@ capability, you can directly subscribe to push notifications for a device:
+
+
+```[android]
+realtime.channels.get("pushenabled:foo").push.subscribeDevice(context);
+
+// or
+
+realtime.channels.get("pushenabled:foo").push.subscribeDeviceAsync(context, new CompletionListener() {
+    @Override
+    public void onSuccess() {}
+
+    @Override
+    public void onError(ErrorInfo errorInfo) {
+        // Handle error.
+    }
+});
+```
+
+```[swift]
+realtime.channels.get("pushenabled:foo").push.subscribeDevice { error
+    // Check error.
+}
+```
+
+```[objc]
+[[realtime.channels get:@"pushenabled:foo"].push subscribeDevice:^(ARTErrorInfo *error) {
+    // Check error.
+}];
+```
+
+h4. Without push-subscribe capabilities
+
+If your application lacks @push-subscribe@ capability, you can register a device for push notifications through the "@activate()@":/api/realtime-sdk/push?langandroid#activate method provided by the @client.push@ object. This method registers the device with the server, and the @deviceId@ will be generated:
+
+```
+client.push.activate();
+```
+
+The "@DeviceDetails@":/api/realtime-sdk/push-admin object contains the @deviceId@, @clientId@, along with other recipient attributes, and it can be accessed once the device is registered.
```

Comment:
I think you've got a little muddled here. This is activation related, not subscription related? 

I think the original content is instead talking about your device sending its `deviceId` to a server and the server uses [this method](https://ably.com/docs/api/realtime-sdk/push-admin#push-channel-sub-save) to subscribe it to a channel.